kani-compiler/src/codegen_cprover_gotoc/codegen/contract.rs

```diff
@@ -37,6 +37,20 @@ impl<'tcx> GotocCtx<'tcx> {
         let function_under_contract_attrs = KaniAttributes::for_item(tcx, function_under_contract);
         let wrapped_fn = function_under_contract_attrs.inner_check().unwrap().unwrap();
 
+        let recursion_tracker = items.iter().find_map(|i| match i {
+            MonoItem::Static(recursion_tracker)
+                if (*recursion_tracker).name().contains("REENTRY") =>
```

Comment:
Is there a way to make this name match more specific? For instance like only searching the siblings of the harness function? There is code in `attributes.rs` that does similar stuff, or you may consider using `TyCtxt::parent_module_from_def_id` to get the module containing the harness and then `TyCtxt::module_children` to get the siblings that you can search.

Also I would suggest using `filter_map` and fail if the resulting iterator has more than one matching element.

Reply:
https://github.com/model-checking/kani/pull/3045#discussion_r1508420857